diff --git a/motion_planners/trajopt/CMakeLists.txt b/motion_planners/trajopt/CMakeLists.txt
index 1111111..2222222 100644
--- a/motion_planners/trajopt/CMakeLists.txt
+++ b/motion_planners/trajopt/CMakeLists.txt
@@ -1,6 +1,9 @@
 find_package(trajopt REQUIRED)
 find_package(trajopt_sco REQUIRED)
 find_package(osqp REQUIRED)
+if(NOT TARGET osqp::osqp AND TARGET osqp::osqpstatic)
+  add_library(osqp::osqp ALIAS osqp::osqpstatic)
+endif()
 
 # Trajopt Planner
 add_library(
