--- a/bindings/python/CMakeLists.txt
+++ b/bindings/python/CMakeLists.txt
@@ -224,7 +224,7 @@ if(BUILD_PYTHON_INTERFACE)
 
   if(BUILD_WITH_CASADI_SUPPORT)
     pinocchio_python_bindings_specific_type(casadi)
-    target_link_libraries(${PYWRAP}_casadi PUBLIC casadi)
+    target_link_libraries(${PYWRAP}_casadi PUBLIC ${CASADI_TARGET})
     list(APPEND STUBGEN_DEPENDENCIES "${PYWRAP}_casadi")
 
     target_precompile_headers(
